huaweicloud/services/cfw/resource_huaweicloud_cfw_lts_log.go

```diff
@@ -0,0 +1,332 @@
+		"lts_flow_log_stream_enable":   d.Get("lts_flow_log_stream_enable"),
+		"lts_attack_log_stream_id":     d.Get("lts_attack_log_stream_id"),
+		"lts_access_log_stream_id":     d.Get("lts_access_log_stream_id"),
+		"lts_flow_log_stream_id":       d.Get("lts_flow_log_stream_id"),
+	}
+}
+
+func resourceLtsLogRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	var (
+		httpUrl = "v1/{project_id}/cfw/logs/configuration"
+		product = "cfw"
+	)
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating CFW Client: %s", err)
+	}
+
+	path := client.Endpoint + httpUrl
+	path = strings.ReplaceAll(path, "{project_id}", client.ProjectID)
+	path += fmt.Sprintf("?fw_instance_id=%s", d.Id())
+
+	opt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	resp, err := client.Request("GET", path, &opt)
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, parseQueryError(err), "error retrieving LTS log configuration")
+	}
+
+	respBody, err := utils.FlattenResponse(resp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	configuration, err := jmespath.Search("data", respBody)
+	if err != nil {
+		diag.Errorf("error parsing data from response= %#v", respBody)
+	}
+
+	ltsEnable := utils.PathSearch("lts_enable", configuration, float64(0)).(float64)
+	if int(ltsEnable) == 0 {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "error retrieving LTS log configuration")
+	}
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("fw_instance_id", utils.PathSearch("fw_instance_id", configuration, nil)),
+		d.Set("lts_log_group_id", utils.PathSearch("lts_log_group_id", configuration, nil)),
+		d.Set("lts_attack_log_stream_id", utils.PathSearch("lts_attack_log_stream_id", configuration, nil)),
+		d.Set("lts_access_log_stream_id", utils.PathSearch("lts_access_log_stream_id", configuration, nil)),
+		d.Set("lts_flow_log_stream_id", utils.PathSearch("lts_flow_log_stream_id", configuration, nil)),
+		d.Set("lts_attack_log_stream_enable", utils.PathSearch("lts_attack_log_stream_enable", configuration, nil)),
+		d.Set("lts_access_log_stream_enable", utils.PathSearch("lts_access_log_stream_enable", configuration, nil)),
+		d.Set("lts_flow_log_stream_enable", utils.PathSearch("lts_flow_log_stream_enable", configuration, nil)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func parseQueryError(err error) error {
+	var errCode golangsdk.ErrDefault400
+	if errors.As(err, &errCode) {
+		var apiError interface{}
+		if jsonErr := json.Unmarshal(errCode.Body, &apiError); jsonErr != nil {
+			return err
+		}
+		errorCode, errorCodeErr := jmespath.Search("error_code", apiError)
+		if errorCodeErr != nil {
+			return err
+		}
+
+		if errorCode == "CFW.00200005" {
+			return golangsdk.ErrDefault404(errCode)
+		}
+	}
+	return err
+}
```

Comment:
this func can be merged with func parseCreateError

Reply:
fixed